### ISSUE TITLE
Failed to import Media: Remote server did not respond - since 0.6.4 on GAE

### ISSUE THREAD

**Issue by marcoherzog** [WordPress/wordpress-importer#38]
I am running a Wordpress multisite on google app engine (flexible environment) with <a href="https://wordpress.org/plugins/gcs/" rel="noopener" target="_blank">Google Cloud Storage plugin</a>.  Since the importer update to 0.6.4 I got errors for media imports. If I go back to 0.6.3 everything works as it should! 

It must be related to the replacement of the depreciated <a href="https://developer.wordpress.org/reference/functions/wp_get_http/" rel="noopener" target="_blank">wp_get_function</a>-function. 

wp_safe_remote_get ([code link](https://github.com/WordPress/wordpress-importer/blob/master/src/wordpress-importer.php#L1000-L1004)) responds with:   

> {errors: {http_request_failed: ["Destination directory for file streaming does not exist or is not writable."]}, error_data: []} 
  
Any help on fixing this in a following version would be really appreciated.

<hr>

For now my **workaround** is to revert the following lines in wordpress-importer.php (beginning line 1000) back to the previous version:
```
$remote_response = wp_safe_remote_get( $url, array(
    'timeout' => 300,
    'stream' => true,
    'filename' => $upload['file'],
) );
$headers = wp_remote_retrieve_headers( $remote_response );
```
change to
```
$headers = wp_get_http( $url, $upload['file'] );
```

AND

```
$remote_response_code = wp_remote_retrieve_response_code( $remote_response );

// make sure the fetch was successful
if ( $remote_response_code != '200' ) {

    @unlink( $upload['file'] );
    return new WP_Error( 'import_file_error', sprintf( __('Remote server returned error response %1$d %2$s', 'wordpress-importer'), esc_html($remote_response_code), get_status_header_desc($remote_response_code) ) );
}
```
change to
```
// make sure the fetch was successful
if ( $headers['response'] != '200' ) {
    @unlink( $upload['file'] );
    return new WP_Error( 'import_file_error', sprintf( __('Remote server returned error response %1$d %2$s', 'wordpress-importer'), esc_html($headers['response']), get_status_header_desc($headers['response']) ) );
}
```

See also my ![wish screenshot](https://user-images.githubusercontent.com/2692054/41724050-f46162e4-756c-11e8-875b-fd4c1a55c75c.png)

**Comment by cyberish** [comment]
Thanks, this solved my problem as well

**Comment by mpacary** [comment]
@marcoherzog Big thanks !

Note that I had to modify as well function `wp_get_http()` into `wp-includes/deprecated.php`

Changed:
`$response = wp_safe_remote_request( $url, $options );`
to
`$response = wp_remote_request( $url, $options );`

Looks like `wp_safe_remote_request()` does not likes URL like `http://<IP address>:<port>/<path>` (??), however that was the only way to access media from an old site.
I was only getting "Remote server did not respond" without any hint like "URL is unsafe"...

**Comment by ocean90** [comment]
This seems to be a file permission error with the temporary directory. The error messages is [triggered by this check](https://github.com/WordPress/WordPress/blob/735d31ffbba2054c55cd7618cb0214da311a4d79/wp-includes/class-http.php#L298-L303). The filename is a path to a temporary file generated [here](https://github.com/WordPress/wordpress-importer/blob/458e9648fa9fe9981cc12eba99777ab42cd03c38/src/class-wp-import.php#L998).

The missing/incorrect error message is fixed in #72.
